isis/tests/FunctionalTestsMsi2isis.cpp

```diff
@@ -0,0 +1,469 @@
+
+  ASSERT_NEAR(hist->Average(), 0.040669880452177, .000001);
+  ASSERT_NEAR(hist->Sum(), 8997.9670307614, .0001);
+  ASSERT_EQ(hist->ValidPixels(), 221244);
+  ASSERT_EQ(hist->NullPixels(), 0);
+  ASSERT_NEAR(hist->StandardDeviation(), 0.017675184374796, .0001);
+}
+
+TEST(Msi2isis, Msi2isisTestCubicConvolution) {
+  Pvl appLog;
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/m0126865998f4_2p_iof.cubic.cub";
+  QVector<QString> args = {"from=data/near/msi2isis/m0126865998f4_2p_iof.lbl",
+                            "to=" + cubeFileName };
+
+  UserInterface options(APP_XML, args);
+  try {
+   std::cerr << "Running msi2isis..." << std::endl;
+   msi2isis(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest NEAR/MSI file: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions Group
+  ASSERT_EQ(cube.sampleCount(), 537);
+  ASSERT_EQ(cube.lineCount(), 412);
+  ASSERT_EQ(cube.bandCount(), 1);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  ASSERT_EQ(inst["SpacecraftName"][0].toStdString(), "NEAR EARTH ASTEROID RENDEZVOUS");
+  ASSERT_EQ(inst["InstrumentId"][0].toStdString(), "MSI" );
+  ASSERT_EQ(inst["TargetName"][0].toStdString(), "EROS" );
+  ASSERT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "126865998830" );
+  ASSERT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "126865998919" );
+  ASSERT_EQ(inst["OriginalSpacecraftClockStartCount"][0].toStdString(), "126865998.830" );
+  ASSERT_EQ(inst["OriginalSpacecraftClockStopCount"][0].toStdString(), "126865998.919" );
+  ASSERT_DOUBLE_EQ(double(inst["ExposureDuration"]), 89.0);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2000-02-25T05:16:12.656" );
+  ASSERT_EQ(inst["StopTime"][0].toStdString(), "2000-02-25T05:16:12.745" );
+  ASSERT_DOUBLE_EQ(double(inst["DpuDeckTemperature"]), 286.5);
+  ASSERT_EQ(inst["DpuDeckTemperature"].unit().toStdString(), "K");
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  ASSERT_EQ(archive["DataSetId"][0].toStdString(), "NEAR-A-MSI-3-EDR-EROS/ORBIT-V1.0");
+  ASSERT_EQ(archive["ProductId"][0].toStdString(), "M0126865998F4_2P_IOF.FIT");
+  ASSERT_EQ(archive["InstrumentHostName"][0].toStdString(), "NEAR EARTH ASTEROID RENDEZVOUS");
+  ASSERT_EQ(archive["InstrumentName"][0].toStdString(), "MULTI-SPECTRAL IMAGER");
+  ASSERT_EQ(archive["InstrumentId"][0].toStdString(), "MSI");
+  ASSERT_EQ(archive["TargetName"][0].toStdString(), "EROS");
+
+  // BandBin Group
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  ASSERT_EQ(int(bandbin["FilterNumber"]), 4);
+  ASSERT_EQ(int(bandbin["Center"]), 950);
+  ASSERT_EQ(bandbin["Center"].unit().toStdString(), "nm");
+
+ // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -93001);
```

Comment:
Again you don't need to re-test all of these label values, just anything that changes when you set cubic convolution.

Reply:
I have left these tests in because if for some reason the input changes, the test will certainly fail. Testing the label will immediately indicate why.

Reply:
The problem is that these lines are re-checking that same values as previous tests but not doing anything that should change them. For each input, we should only check that their labels are correct one time. So, it is okay to have one test check that the old label is correct and one test check that the new label is correct. We should not be checking that in every test that uses them.